<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModelOffset</class>
 <widget class="qMRMLWidget" name="ModelOffset">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>442</width>
    <height>692</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Shift models</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_toolTipTransform">
        <property name="text">
         <string>Tool Tip Transform:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="toolTipTransformSelector">
        <property name="toolTip">
         <string>Pick the transform of the tip of the tool.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLLinearTransformNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_controlPointsTransform">
        <property name="text">
         <string>Control Points:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="controlPointsSelector">
        <property name="toolTip">
         <string>Pick the markups of the ureter.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLMarkupsFiducialNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_appliedModels">
        <property name="text">
         <string>Apply to Models:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLSubjectHierarchyTreeView" name="appliedModelsTreeView">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="dragEnabled">
         <bool>false</bool>
        </property>
        <property name="alternatingRowColors">
         <bool>false</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::MultiSelection</enum>
        </property>
        <property name="showRootItem">
         <bool>true</bool>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true"/>
        </property>
        <property name="visibilityColumnVisible">
         <bool>true</bool>
        </property>
        <property name="idColumnVisible">
         <bool>false</bool>
        </property>
        <property name="colorColumnVisible">
         <bool>false</bool>
        </property>
        <property name="excludeNodeAttributeNamesFilter">
         <stringlist/>
        </property>
        <property name="attributeNameFilter">
         <string/>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="advancedCollapsibleButton">
     <property name="text">
      <string>Advanced</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_shift_x">
        <property name="text">
         <string>Apply left-right (X) offset: </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="offsetXCheckBox">
        <property name="toolTip">
         <string>If checked, model will be shifted along transverse (X) axis</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_shift_y">
        <property name="text">
         <string>Apply postero-anterior (Y) offset: </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="offsetYCheckBox">
        <property name="toolTip">
         <string>If checked, model will be shifted along sagittal (Y) axis</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_shift_z">
        <property name="text">
         <string>Apply supero-inferior (Z) offset: </string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="offsetZCheckBox">
        <property name="toolTip">
         <string>If checked, model will be shifted along vertical (Z) axis</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="applyButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>Run the algorithm.</string>
     </property>
     <property name="text">
      <string>Apply offset - shift selected models</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="applyUndoButton">
     <property name="text">
      <string>Remove offset form selected models</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSubjectHierarchyTreeView</class>
   <extends>QTreeView</extends>
   <header>qMRMLSubjectHierarchyTreeView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>ModelOffset</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>toolTipTransformSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>250</x>
     <y>81</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ModelOffset</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>controlPointsSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>244</y>
    </hint>
    <hint type="destinationlabel">
     <x>250</x>
     <y>110</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ModelOffset</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>appliedModelsTreeView</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>220</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
